pkg/function/kube_task.go

```diff
@@ -64,6 +68,9 @@ func kubeTask(ctx context.Context, cli kubernetes.Interface, namespace, image st
 		Command:      command,
 		PodOverride:  podOverride,
 	}
+	// Mark labels to pods with prefix `kanister.io`. Add the jobID as reference to the origin for the pod.
+	validateFunc := validateLabelKeyIsPresentFunc(consts.LabelPrefix)
+	kube.AddLabelsToPodOptionsFromContext(ctx, options, path.Join(consts.LabelPrefix, jobIDSuffix), validateFunc)
```

Comment:
@mabhi I didn't mean passing validateFunc as an arg to the function. The `kube.AddLabelsToPodOptionsFromContext()` util function should just do one thing, add labels to the options from context. The validation part should be done separately, before calling that function.

Reply:
Simplied the validation and add-label-to-pod-options functions.